hangman.rb

```diff
@@ -0,0 +1,262 @@
+
+class Hangman
+  def initialize
+    @game_word = pick_word
+    @wrong_letters = []
+    @errors = 0
+
+    # initializes array_of_slots with underscores
+    @num_of_slots = @game_word.length
+    @array_of_slots = []
+    @num_of_slots.times do
+      @array_of_slots.push("_ ")
+    end
+
+    @drawings = [[
+              "|     _________",
+              "|     |/      |",
+              "|     |      ",
+              "|     |      ",
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+            [
+              "|     _________",
+              "|     |/      |",
+              "|     |      " + "(_)".red,
+              "|     |      ",
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+
+                        [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |       " + "|".red,
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                    [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |       |",
+              "|     |       " + "|".red,
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                    [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      "+ "\\".red + "|",
+              "|     |       |",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|" + "/".red,
+              "|     |       |",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/ ",
+              "|     |       | ",
+              "|     |      " + "/ ".red,
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/",
+              "|     |       |",
+              "|     |      / " + "\\ ".red,
+              "|     |",
+              "| ____|___"
+]]
+
+    start
+  end
+
+  def pick_word
+    word_list = %w{cat dog egg horse donkey elephant pig monkey iguana cow goat unigoat eggplant mouse dragon phoenix}
```

Comment:
This would be a good place to use a constant now that we've used them a bit more